src/Core/Util/String/StringModifier.php

```diff
@@ -57,4 +64,39 @@ public function cutEnd(string $string, int $expectedLength): string
 
         return $string;
     }
+
+    /**
+     * Return a friendly url made from the provided string
+     * If the mbstring library is available, the output is the same as the js function of the same name.
+     *
+     * @param string $string
+     * @param bool $allowAccentedChars
+     *
+     * @return string
+     */
+    public function str2url(string $string, bool $allowAccentedChars): string
+    {
+        if ($string === '') {
+            return '';
+        }
+
+        $return_str = trim($string);
+        $return_str = mb_strtolower($return_str, 'UTF-8');
+
+        // Remove all non-whitelist chars.
+        if ($allowAccentedChars) {
+            $return_str = preg_replace('/[^a-zA-Z0-9\s\'\:\/\[\]\-\p{L}]/u', '', $return_str);
+        } else {
+            if (null === $this->transliterator) {
+                $this->transliterator = Transliterator::create('Any-Latin; Latin-ASCII');
+            }
+
+            $return_str = $this->transliterator->transliterate($string);
+            $return_str = preg_replace('/[^a-zA-Z0-9\s\'\:\/\[\]\-]/', '', $return_str);
```

Comment:
Why do not try to move also replaceAccentedChars function ?

Reply:
Done !